doc/development/deprecations.rst

```diff
@@ -34,6 +34,20 @@ Pending deprecations
 
   - Added and deprecated for ``Sum`` and ``Prod`` instances in v0.35
 
+* Accessing ``qml.ops.Hamiltonian`` with new operator arithmetic is deprecated. Using ``qml.Hamiltonian``
+  with new operator arithmetic enabled now returns a ``LinearCombination`` instance. Some functionality
```

Comment:
Do we want to hide the fact that the new returned operator is `LinearCombination` from users?